apps/passport-client/src/validateState.ts

```diff
@@ -0,0 +1,174 @@
+import { User, requestLogToServer } from "@pcd/passport-interface";
+import { PCDCollection } from "@pcd/pcd-collection";
+import {
+  SemaphoreIdentityPCD,
+  SemaphoreIdentityPCDPackage
+} from "@pcd/semaphore-identity-pcd";
+import { appConfig } from "./appConfig";
+import { AppState } from "./state";
+
+/**
+ * Determines whether the app's global state as represented by {@link AppState} object
+ * contains valid data. If it does not, returns the set of things that are incorrect about
+ * it in an array of human interpretable strings. If there are no errors, returns an
+ * empty array.
+ */
+export function validateState(state: AppState): string[] {
+  const validationErrors: string[] = [];
+
+  if (!state.self) {
+    validationErrors.push("missing 'self' field from app state");
```

Comment:
Given where this is called, I think it will always reject a logged-out state as invalid.
There need to be separate validators (or branches in this validator) for logged-in vs. logged-out.
I'd actually like to validate that we never get any partially-logged-in states (e.g. self set but identity not set).